dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -147,11 +160,26 @@ public CommunityRest findOne(Context context, UUID id) {
     @Override
     public Page<CommunityRest> findAll(Context context, Pageable pageable) {
         try {
-            long total = cs.countTotal(context);
-            List<Community> communities = cs.findAll(context, pageable.getPageSize(),
+            if (authorizeService.isAdmin(context)) {
+                long total = cs.countTotal(context);
+                List<Community> communities = cs.findAll(context, pageable.getPageSize(),
                     Math.toIntExact(pageable.getOffset()));
-            return converter.toRestPage(communities, pageable, total, utils.obtainProjection());
-        } catch (SQLException e) {
+                return converter.toRestPage(communities, pageable, total, utils.obtainProjection());
+            } else {
+                List<Community> communities = new LinkedList<Community>();
+                DiscoverQuery discoverQuery = new DiscoverQuery();
```

Comment:
Same here, a comment to describe the query.  This looks to be searching for all Communities which the current user has access to.

Reply:
done here https://github.com/DSpace/DSpace/pull/2707/commits/9ce5c586269ebdde32e87d1d7591be15734f7dd3#diff-38aef90b55ad47e2a09766c3dedbf89dR169